File: Python/Math-Array-String/75.py
# 387. First Unique Character in a String


class Solution:
    def firstUniqChar(self, s: str) -> int:
        char_count = {}

        # Count frequency of each character
        for char in s:
            char_count[char] = char_count.get(char, 0) + 1

        # Find the first non-repeating character
        for index, char in enumerate(s):
            if char_count[char] == 1:
                return index

        return -1
    
sol = Solution()
output = sol.firstUniqChar("leetcode")
print(output)  # Output: 0

File: JavaScript/Math-Array-String/75.js
// 387. First Unique Character in a String

/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
    let charCount = new Map();

    // Count frequency of each character
    for (let char of s) {
        charCount.set(char, (charCount.get(char) || 0) + 1);
    }

    // Find the first non-repeating character
    for (let i = 0; i < s.length; i++) {
        if (charCount.get(s[i]) === 1) {
            return i;
        }
    }

    return -1;
};

// Test cases
console.log(firstUniqChar("leetcode"));      // Output: 0
console.log(firstUniqChar("loveleetcode"));  // Output: 2
console.log(firstUniqChar("aabb"));          // Output: -1

